src/resources/js/dependency.js

```diff
@@ -72,6 +75,7 @@
 						}
 					} else {
 						$dependent
+							.hide()
```

Comment:
and I've not removed the class, just use inline styles set by JS to make sure what is hidden is hidden and what should show shows